src/expr.rs

```diff
@@ -87,129 +134,273 @@ pub struct Prefix<'a> {
 /// Create a reference to a field (`field`) of a TCP option header (`name`).
 pub struct TcpOption<'a> {
+    /// The TCP option header name.
     pub name: Cow<'a, str>,
-    pub field: Cow<'a, str>,
+    /// The field name.
+    ///
+    /// If the field property is not given, the expression is to be used as a
+    /// TCP option existence check in a match statement with a boolean on the
+    /// right hand side.
+    pub field: Option<Cow<'a, str>>,
+}
+
+/// Default TCP option for `maxseg` option.
+impl Default for TcpOption<'_> {
+    fn default() -> Self {
+        TcpOption {
+            name: "maxseg".into(),
+            field: None,
+        }
+    }
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(rename = "sctp chunk")]
 /// Create a reference to a field (`field`) of an SCTP chunk (`name`).
 pub struct SctpChunk<'a> {
+    /// The SCTP chunk name.
     pub name: Cow<'a, str>,
+    /// The field name.
+    ///
+    /// If the field property is not given, the expression is to be used as an
+    /// SCTP chunk existence check in a match statement with a boolean on the
+    /// right hand side.
     pub field: Cow<'a, str>,
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(rename = "meta")]
 /// Create a reference to packet meta data.
+///
+/// See also: <https://wiki.nftables.org/wiki-nftables/index.php/Matching_packet_metainformation>
 pub struct Meta {
+    /// The packet [meta data key](MetaKey).
     pub key: MetaKey,
 }
 
+/// Default impl for meta key `l4proto`.
+impl Default for Meta {
+    fn default() -> Self {
+        Meta {
+            key: MetaKey::L4proto,
+        }
+    }
+}
+
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, Serialize, Deserialize)]
 #[serde(rename_all = "lowercase")]
 /// Represents a `meta` key for packet meta data.
+///
+/// See also: <https://wiki.nftables.org/wiki-nftables/index.php/Matching_packet_metainformation>
 pub enum MetaKey {
+    // matching by packet info:
+    /// Packet type (unicast, broadcast, multicast, other).
+    Pkttype,
+    /// Packet length in bytes.
     Length,
+    /// Packet protocol / EtherType protocol value.
     Protocol,
-    Priority,
-    Random,
-    Mark,
+    /// Netfilter packet protocol family.
+    Nfproto,
+    /// Layer 4 protocol.
+    L4proto,
+
+    // matching by interface:
+    /// Input interface index.
     Iif,
+    /// Input interface name.
     Iifname,
+    /// Input interface type.
     Iiftype,
+    /// Input interface kind name.
+    Iifkind,
+    /// Input interface group.
+    Iifgroup,
+    /// Output interface index.
     Oif,
+    /// Output interface name.
     Oifname,
+    /// Output interface type.
     Oiftype,
-    Skuid,
-    Skgid,
-    Nftrace,
-    Rtclassid,
-    Ibriport,
-    Obriport,
+    /// Output interface kind name.
+    Oifkind,
+    /// Output interface group.
+    Oifgroup,
+    /// Input bridge interface name.
     Ibridgename,
```

Comment:
Hmm, should these rather be sorted alphabetically?

Reply:
I grouped/sorted them by category of match subject and I think this will do for now.